File: python-samples/log_line_count_with_line_size_sum.py
import sys

log_file_path = sys.argv[1]

log_line_count = 0
line_size_sum = 0

with open(log_file_path, 'r') as log_file:
    while line := log_file.readline():
        log_line_count = log_line_count + 1
        line_size_sum += len(line.rstrip('\n'))

print("log line count:", log_line_count)
print("log line size sum:", line_size_sum)

File: java-samples/src/main/java/posts/parthmistry/javasamples/LogLineCountWithLineSizeSum.java
package posts.parthmistry.javasamples;

import java.nio.file.Files;
import java.nio.file.Paths;

public class LogLineCountWithLineSizeSum {

    public static void main(String[] args) throws Exception {
        var logFilePath = args[0];

        var logLineCount = 0L;
        var lineSizeSum = 0L;

        try (var reader = Files.newBufferedReader(Paths.get(logFilePath))) {
            String line = null;
            while ((line = reader.readLine()) != null) {
                logLineCount += 1;
                lineSizeSum += line.length();
            }
        }

        System.out.println("log line count: " + logLineCount);
        System.out.println("log line size sum: " + lineSizeSum);
    }

}
